docs/Maths.md

```diff
@@ -226,6 +226,107 @@ Result:
+
+### Using cairosvg (*A faster and efficient implementation*) ### 
+
+A faster and expected approach of embedding a `Pygal` svg graph into a PDF file is to use the `cairosvg` library to convert the `svg` string generated by `pygal` into byte string using `BytesIO` library so that we can keep these data in an in-memory buffer. 
+As the `fpdf` library can understand byte string of a `svg` content, it can easily embed a graph inside a `pdf`. 
+
+```python
+import pygal
+from fpdf import FPDF
+from io import BytesIO
+import cairosvg
+
+# Create a Pygal bar chart
+bar_chart = pygal.Bar()
+bar_chart.title = 'Browser usage evolution (in %)'
+bar_chart.x_labels = map(str, range(2002, 2013))
+bar_chart.add('Firefox', [None, None, 0, 16.6,   25,   31, 36.4, 45.5, 46.3, 42.8, 37.1])
+bar_chart.add('Chrome',  [None, None, None, None, None, None,    0,  3.9, 10.8, 23.8, 35.3])
+bar_chart.add('IE',      [85.8, 84.6, 84.7, 74.5,   66, 58.6, 54.7, 44.8, 36.2, 26.6, 20.1])
+bar_chart.add('Others',  [14.2, 15.4, 15.3,  8.9,    9, 10.4,  8.9,  5.8,  6.7,  6.8,  7.5])
+
+# Use CairoSVG to convert PNG from SVG of barchart
+svg_img_byte = BytesIO()
+cairosvg.svg2png(bar_chart.render(), write_to=svg_img_byte)
+
+# Set the position and size of the image in the PDF
+x = 50
+y = 50
+w = 100
+h = 70
+
+# Make the PDF
+pdf = FPDF()
+pdf.add_page()
+pdf.image(svg_img_byte, x=x, y=y, w=w, h=h)
+pdf.output('bar_chart.pdf')
+```
+The above code generates a pdf with title `bar_chart.pdf` file with following graph -
+![](pygal_chart_cairo.png)
+
+**!! Troubleshoot: !!**
+
+You may encounter `GTK` (Gnome Toolkit) errors while executing the above example in windows. Error could be like following -
+```
+OSError: no library called "cairo-2" was found
+no library called "cairo" was found
+no library called "libcairo-2" was found
+cannot load library 'libcairo.so.2': error 0x7e
+cannot load library 'libcairo.2.dylib': error 0x7e
+cannot load library 'libcairo-2.dll': error 0x7e
+```
+In this case install install `GTK` from [GTK-for-Windows-Runtime-Environment-Installer](https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer/releases). Restart your editor. And you are all done.
+
+### Using svglib and reportlab (*A slower and purely pythonic implementation*) ### 
+```python
+import pygal
+from reportlab.graphics import renderPM
+from svglib.svglib import svg2rlg
+from fpdf import FPDF
+
+# Create a Pygal bar chart
+bar_chart = pygal.Bar()
+bar_chart.title = 'Sales by Year'
+bar_chart.x_labels = ['2016', '2017', '2018', '2019', '2020']
+bar_chart.add('Product A', [500, 750, 1000, 1250, 1500])
+bar_chart.add('Product B', [750, 1000, 1250, 1500, 1750])
+
+# Render the chart to an SVG and PNG file
+bar_chart.render_to_file('bar_chart.svg')
+drawing = svg2rlg("bar_chart.svg")
+renderPM.drawToFile(drawing, "bar_chart.png")
```

Comment:
```suggestion
svg_img = bar_chart.render()
# Code inspired from: https://github.com/deeplook/svglib/blob/v1.5.1/svglib/svglib.py#L1446
xml_parser = etree.XMLParser(remove_comments=True, recover=True, resolve_entities=resolve_entities)
svg_root = etree.parse(path, parser=xml_parser).getroot()  # from lxml import etree
# Code inspired by: https://github.com/deeplook/svglib/blob/v1.5.1/svglib/svglib.py#L1415
drawing = SvgRenderer('').render(svg_root)  # from svglib.svglib import SvgRenderer
img_bytes = renderPM.drawToString(drawing)
```
Couldn't this be done fully in-memory?

Reply:
This is clever. Did not think or explore like this. 
I will run it and will let you notify about the verdict.

Reply:
@Lucas-C , I tried your suggestions. Here are what I found -

- `PyGal` generates SVG strings (XML strings) using the `render()` method.
- The `parse()` method of `lxml` element tree parsing technique can not parse the SVG string as the implementation of `parse()` accepts a `file path` or a `pathlike` object as a parameter instead of an SVG string.
- Instead, I used the `fromstring()` method provided by `lxml` to achieve in-memory SVG parsing.
- The `fpdf.image()` method use `'utf-8' codec` whereas the `renderPM.drawToString()` method returns an image string that might not be a 'utf-8' encoded string which returns following error.
```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x87 in position 10: invalid start byte
```
I tried to google the error. Turns out the image encoding need to be done in `utf-8` method. And from `fpdf.image()` method's end we can simply ignore that byte which of course not a bright idea here. Link: [Here](https://stackoverflow.com/questions/12468179/unicodedecodeerror-utf8-codec-cant-decode-byte-0x9c). There maybe other approaches to address this issue. I did not dig deeper into this as of now.
- To get rid of this problem we may use the `io.BytesIO` library to make it a Byte string that is readable by the `image()` method. 

Accommodating all the changes of the abovementioned discussion it boils down to the following snippet-

```python
svg_img = bar_chart.render()
xml_parser = etree.XMLParser()
svg_root = etree.fromstring(svg_img, parser=xml_parser)
drawing = SvgRenderer(svg_img).render(svg_root)
drawing_img_byte = renderPM.drawToString(drawing)
img_bytes = io.BytesIO(drawing_img_byte)
```

We can embed charts using this in-memory technique. Please let me know what do you think?

Reply:
You did a really good job of exploring & experimenting overall! 🙏

`etree.fromstring()` was a really good idea. If it all works with a "bare" `XMLParser` instance, maybe the `parser` argument can be fully omitted? _E.g._ `svg_root = etree.fromstring(svg_img)`. _cf._ [documentation](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.fromstring)

Apart from that minor suggestion, I think this code would be perfect for our new documentation section 😊